feat(p2p/dht): Enable DHT server metrics

crStiv opened the pull request:
### Description:
This PR implements DHT server metrics to track server load and performance. Added metrics include:

### Key metrics:
- peers_total: Total number of connected peers
- requests_total: Total number of DHT requests by type
- request_duration_seconds: Duration of DHT requests
- routing_table_size: Size of routing table
- stored_records_total: Total number of stored records

All metrics are available via Prometheus endpoint with `celestia_dht_` prefix and include network/node_type labels.

### Closes #4066

walldiss commented:
Hey @crStiv! Thank you for taking this one! I see that you made wrappers and implemented metrics manually. Could you please consider using metrics that are already implemented inside libp2p/kad-dht pkg?

crStiv commented:
> Hey @crStiv! Thank you for taking this one! I see that you made wrappers and implemented metrics manually. Could you please consider using metrics that are already implemented inside libp2p/kad-dht pkg? You would need to plug them to our metrics collector

Hi Vlad, I've updated dht.go, does it seem good for you?

walldiss commented:
Changes does not make any sense. Code will not compile and most of introduced code in the PR is not even called.
